src/plugins/data_source_management/public/components/data_source_error_menu/data_source_error_menu.tsx

```diff
@@ -3,14 +3,86 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { EuiIcon, EuiText } from '@elastic/eui';
-import React from 'react';
+import React, { useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiText,
+} from '@elastic/eui';
+import { ErrorIcon } from '../custom_database_icon';
+import { DataSourceDropDownHeader } from '../drop_down_header';
+
+interface DataSourceErrorMenuProps {
+  application?: ApplicationStart;
+}
+
+export const DataSourceErrorMenu = ({ application }: DataSourceErrorMenuProps) => {
+  const [showPopover, setShowPopover] = useState<boolean>(false);
+
+  const refreshButton = (
+    <EuiButton
+      data-test-subj="dataSourceErrorRefreshButton"
+      fill={false}
+      iconType={'refresh'}
+      size="s"
+      onClick={() => window.location.reload()}
+    >
+      {i18n.translate('dataSourcesManagement.dataSourceErrorMenu.refreshPage', {
+        defaultMessage: 'Refresh the page',
+      })}
+    </EuiButton>
+  );
+
+  const iconButton = (
+    <EuiButtonIcon
+      className="euiHeaderLink"
+      data-test-subj="dataSourceErrorMenuHeaderLink"
+      aria-label={i18n.translate('dataSourceError.dataSourceErrorMenuHeaderLink', {
+        defaultMessage: 'dataSourceErrorMenuHeaderLink',
+      })}
+      iconType={() => <ErrorIcon />}
+      size="s"
+      onClick={() => setShowPopover(!showPopover)}
+    />
+  );
 
-export const DataSourceErrorMenu = () => {
   return (
     <>
-      <EuiIcon type={'crossInCircleFilled'} color={'danger'} />
-      <EuiText color={'danger'}>Error</EuiText>
+      <EuiPopover
+        id={'dataSourceErrrorPopover'}
```

Comment:
typo `dataSourceErrrorPopover`, it should `dataSourceErrorPopover`

Reply:
Good catch! Will fix!